docs/source/examples/de_sscan.ipynb

```diff
@@ -0,0 +1,1505 @@
+    "both detectors and the motor.  The `sscan` record has buffers capable of\n",
+    "collecting as many as 1,000 points per array.  First get the number of points\n",
+    "collected, then limit each array length to that number.\n",
+    "\n",
+    "If we write this as a function, we can call it again later.  Since it executes quickly, it does not need to be written as a bluesky plan."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'m1': array([-1.2 , -1.08, -0.96, -0.84, -0.72, -0.6 , -0.48, -0.36, -0.24,\n",
+       "        -0.12,  0.  ,  0.12,  0.24,  0.36,  0.48,  0.6 ,  0.72,  0.84,\n",
+       "         0.96,  1.08,  1.2 ]),\n",
+       " 'I0': array([1., 2., 2., 1., 1., 1., 1., 0., 1., 3., 4., 3., 2., 1., 0., 1., 1.,\n",
+       "        2., 1., 1., 1.], dtype=float32),\n",
+       " 'diode': array([1., 1., 1., 1., 1., 0., 1., 2., 2., 3., 3., 3., 3., 1., 1., 1., 1.,\n",
+       "        1., 1., 1., 1.], dtype=float32)}"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "def get_sscan_data():\n",
+    "    npts = scan1.current_point.get()\n",
+    "    data = {\n",
+    "        \"m1\": scan1.positioners.p1.array.get()[:npts],\n",
+    "        \"I0\": scan1.detectors.d01.array.get()[:npts],\n",
+    "        \"diode\": scan1.detectors.d02.array.get()[:npts],\n",
+    "    }\n",
+    "    return data\n",
+    "\n",
+    "get_sscan_data()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Setup and run same scan from bluesky\n",
+    "\n",
+    "Repeat the scan from the previous examples, but use bluesky to configure\n",
+    "`scan1`.  It will be useful to connect the motor, the scaler, and two the scaler\n",
+    "channels."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ophyd import EpicsMotor\n",
+    "from ophyd.scaler import ScalerCH\n",
+    "\n",
+    "m1 = EpicsMotor(f\"{IOC}m1\", name=\"m1\")\n",
+    "scaler1 = ScalerCH(f\"{IOC}scaler1\", name=\"scaler1\")\n",
+    "m1.wait_for_connection()\n",
+    "scaler1.wait_for_connection()\n",
+    "\n",
+    "# for convenience\n",
+    "I0 = scaler1.channels.chan02.s\n",
+    "diode = scaler1.channels.chan04.s"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can supply the count time per point and scan range parameters as arguments to\n",
+    "the setup.  After setting the counting time for the scaler, the next step in the\n",
+    "setup is to clear any existing configuration of `scan1` using its `reset()`\n",
+    "method.\n",
+    "\n",
+    "In a bluesky plan, we'll need to use `apstools.plans.run_blocking_function` with\n",
```

Comment:
Why?

Reply:
above?

Reply:
As we discussed, `.reset()` was written as an ophyd function for convenient command-line use.  This one does many, many `caput` operations.  Each of those take some time and would block the `RE`.